localstripe/resources.py

```diff
@@ -1471,6 +1476,45 @@ def __init__(self, subscription=None, plan=None, quantity=1,
 
         self._subscription = subscription
 
+    def _calculate_amount(self):
+        if self.plan.billing_scheme == 'per_unit':
+            return self.plan.amount * self.quantity
+
+        if self.plan.tiers_mode == 'volume':
+            index = next(
+                (i for i, t in enumerate(self.plan.tiers)
+                    if t['up_to'] == 'inf'
+                    or int(t['from']) <= self.quantity <= int(t['up_to']))
+            )
+            return self._calculate_amount_in_tier(
+                self.quantity, index
+                )
+
+        if self.plan.tiers_mode == 'graduated':
+            quantity = self.quantity
+            amount = 0
+
+            for i, t in enumerate(self.plan.tiers):
+                if quantity <= 0 or int(t['from']) > quantity:
+                    break
+
+                amount += self._calculate_amount_in_tier(
+                    quantity - int(t['from']), i
+                    )
+
+                if t['up_to'] == 'inf':
+                    quantity = 0
+                else:
+                    quantity -= int(t['up_to'])
+
+            return amount
+
+        return 0
+
+    def _calculate_amount_in_tier(self, quantity, index):
+        t = self.plan.tiers[index]
+        return int(t["unit_amount"])*quantity + int(t["flat_amount"])
+
```

Comment:
This could be simplified if with an extra `Tier` class that could be extended to cover the different cases, but will do for now. @adrienverge please if you see something that is not very idiomatic I'm let me know so I can change it :)

Reply:
You're right! But I'd rather not create a `Tier` class: the localstripe API relies on the fact that `StripeObject`s are gettable, listable, updatable... which is not the case with tiers.

Reply:
In this case `Tier` wouldn't inherit from `StripeObject`, it would be just a Value Object that could be used to make the code clearer. We can revisit that idea in the future if new logic that varies upon the type of tiers is added, for now it's simple enough